src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,627 @@
+    UT_LOG_INFO("Calling dsSetFPState(IN:Indicator:[0x%d], IN:State:[%d]" \
+                                                ,eIndicator,uState);
+    status = dsSetFPState((dsFPDIndicator_t)eIndicator,(dsFPDState_t)uState);
+    UT_LOG_INFO("Result dsSetFPState()dsError_t:[%s]",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief This test provides a scope to get Front panel LED state.
+*
+* This test case provides a scope to  get the State of FP indicator to ON and OFF.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 3 and provide indicator index.
+*
+*/
+void test_3_dsFPD_hal_GetFPState(void)
+{
+    gTestID = 3;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsFPDState_t eState = 0;
+
+    selectIndicator();
+    UT_LOG_INFO("Calling dsGetFPState(IN:Indicator:[0x%d], OUT:State)" \
+                                                ,eIndicator);
+    status = dsGetFPState((dsFPDIndicator_t)eIndicator,&eState);
+    UT_LOG_INFO("Result dsGetFPState(IN:Indicator:[0x%d], OUT:State:[%s]) dsError_t:[%s])" \
+                                                ,eIndicator,UT_Control_GetMapString(dsFrontPanelStateTable, eState) \
+                                                , UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+    /* Check that the Indicator is valid */
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Front panel LED Blink functionality.
+*
+* This test case provides a scope to  set the Blink duration and iteration 
+* of FP indicator.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel idicator state should be set to On through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 4 and provide indicator type, Blink duration
+* and Blink iteration.
+*
+*/
+void test_4_dsFPD_hal_SetFPBlink(void)
+{
+    gTestID = 4;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    uint32_t uBlinkDuration;
+    uint32_t uBlinkIterations;
+
+    selectIndicator();
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Enter Blink Duration in ms: ");
+    scanf("%d", &uBlinkDuration);
+    readAndDiscardRestOfLine(stdin);
+
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Enter Blink iteration: ");
+    scanf("%d", &uBlinkIterations);
```

Comment:
It's important to use the correct format specifier (%u) to ensure that the input is interpreted as an unsigned integer and not as a signed integer (%d), which would result in incorrect behavior for values outside the signed integer range.  Here the input to the API is unsigned int, so use the appropriate type during scanf.  Check all through the file for similar conditions.

Reply:
updated